src/systems/ec/implementations/vanilla/packages/ec-component-gallery/ec-component-gallery.js

```diff
@@ -250,6 +290,73 @@ export class Gallery {
     if (this.isDialogSupported && this.overlay) {
       this.overlay.removeEventListener('close', this.handleClickOnCloseButton);
     }
+
+    if (this.attachResizeListener) {
+      window.removeEventListener('resize', this.handleResize);
+    }
+  }
+
+  /**
+   * Check if current display is desktop or mobile
+   */
+  checkScreen() {
+    if (window.innerWidth > this.breakpointMd) {
+      this.isDesktop = true;
+      return;
+    }
+
+    this.isDesktop = false;
+  }
+
+  /**
+   * Hide several gallery items by default
+   * - 2 "lines" of items on desktop
+   * - only 3 items on mobile
+   */
+  hideItems() {
+    if (!this.viewAll) return;
+
+    if (this.isDesktop) {
+      const galleryY = this.element.getBoundingClientRect().y;
```

Comment:
about this @emeryro , the article i was referring to was only warning us about the usage of the `y` in getboundingclientRect(), `top` is more broser compatible, he said, and it does the same thing.

> getBoundingClientRect is reliably supported on all desktop and mobile browsers (with the exclusion of the x and y values which is unstable in IE/Edge and Safari)